lib/Db/TotpSecretMapper.php

```diff
@@ -29,30 +31,30 @@
 
 class TotpSecretMapper extends Mapper {
 
-    public function __construct(IDBConnection $db) {
-        parent::__construct($db, 'twofactor_totp_secrets');
-    }
-
-    /**
-     * @param IUser $user
-     * @throws DoesNotExistException
-     * @return TotpSecret
-     */
-    public function getSecret(IUser $user) {
-        /* @var $qb IQueryBuilder */
-        $qb = $this->db->getQueryBuilder();
-
-        $qb->select('id', 'user_id', 'secret', 'state')
-                ->from('twofactor_totp_secrets')
-                ->where($qb->expr()->eq('user_id', $qb->createNamedParameter($user->getUID())));
-        $result = $qb->execute();
-
-        $row = $result->fetch();
-	$result->closeCursor();
-        if ($row === false) {
-            throw new DoesNotExistException('Secret does not exist');
-        }
-        return TotpSecret::fromRow($row);
-    }
+	public function __construct(IDBConnection $db) {
+		parent::__construct($db, 'twofactor_totp_secrets');
+	}
+
+	/**
+	 * @param IUser $user
+	 * @throws DoesNotExistException
+	 * @return TotpSecret
+	 */
+	public function getSecret(IUser $user): TotpSecret {
+		/* @var $qb IQueryBuilder */
+		$qb = $this->db->getQueryBuilder();
+
+		$qb->select('id', 'user_id', 'secret', 'state')
```

Comment:
@MorrisJobke @nickvergessen @rullzer scrutinizer reports that `select` only takes a single parameter, which is what the phpdoc says: https://github.com/nextcloud/server/blob/fac0a11576874e25141e34749e7bc64a688b2506/lib/public/DB/QueryBuilder/IQueryBuilder.php#L300 Any idea how we could improve this documentation? Would it make sense to switch to [variable length parameter lists](https://github.com/nextcloud/server/blob/fac0a11576874e25141e34749e7bc64a688b2506/lib/private/DB/QueryBuilder/QueryBuilder.php#L386)? It would make the [implementation](https://github.com/nextcloud/server/blob/fac0a11576874e25141e34749e7bc64a688b2506/lib/private/DB/QueryBuilder/QueryBuilder.php#L386) less magical.

Reply:
Mmmmh ... I would vote for only one of the two ways, but as this is public API now we need to keep it stable.

Reply:
variable length parameter lists wouldn't actually break the API. It would behave just like before.

Reply:
Sounds sane @ChristophWurst.

@MorrisJobke it wouldn't break public API. As it accepts the exact same as before. Actually it would make the public API more correct.

Reply:
See https://github.com/nextcloud/server/pull/8796 for a first step